cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -664,3 +668,32 @@ fn extract_error_string<T>(simulation_result: &Result<T>, go_storage: &GoLedgerS
         }
     }
 }
+
+// xdr_to_json takes in a string name of an XDR type in the Stellar Protocol
+// (i.e. from the stellar_xdr crate) as well as a raw byte structure and returns a
+// JSONified string of said structure.
+#[no_mangle]
+pub extern "C" fn xdr_to_json(typename: *mut libc::c_char, xdr: CXDR) -> *mut libc::c_char {
+    let type_str = from_c_string(typename);
+    let the_type = match xdr::TypeVariant::from_str(&type_str) {
+        Ok(t) => t,
+        Err(e) => {
+            return string_to_c(format!(r#"{{ "error": "{e}", "type": "{type_str}" }}"#));
+        }
+    };
+
+    let xdr_bytearray = from_c_xdr(xdr);
+    let mut buffer = xdr::Limited::new(xdr_bytearray.as_slice(), DEFAULT_XDR_RW_LIMITS.clone());
```

Comment:
Self-review: given that we know the size of the structure here (passed as `xdr.len`), would it make sense to make that the maximum limit read length for this and just use `DEFAULT_XDR_RW_LIMITS` for depth? cc @leighmcculloch or maybe @sisuresh? I'm not sure of the security/resource implications here.